src/app/cli/src/graphql.ml

```diff
@@ -352,18 +414,72 @@ module Make (Commands : Coda_commands.Intf) = struct
           Option.map (Program.snark_worker_key coda) ~f:(fun k ->
               (k, Program.snark_work_fee coda) ) )
 
+    let hi transaction_database public_key cursor num_to_query
+        queried_transaction has_previous_page has_next_page =
+      let page_info =
+        {Types.Pagination.Page_info.has_previous_page; has_next_page}
+      in
+      let total_count =
+        Option.value_exn
+          (Transaction_database.get_total_transactions transaction_database
+             public_key)
+      in
+      { Types.Pagination.Connection.edges= queried_transaction
+      ; page_info
+      ; total_count }
+
     let payments =
-      result_field "payments"
-        ~doc:"Payments that a user with public key KEY sent or received"
-        ~args:Arg.[arg "publicKey" ~typ:Types.Input.payment_filter_input]
-        ~typ:(non_null @@ list @@ non_null Types.payment)
-        ~resolve:(fun {ctx= coda; _} () public_key ->
-          let open Result.Let_syntax in
-          let%map public_key =
-            result_of_exn Public_key.Compressed.of_base64_exn public_key
-              ~error:"publicKey address is not valid."
+      io_field "payments"
+        ~args:
+          Arg.
+            [ (* TODO: deal with case where payment filter can be null *)
```

Comment:
Afaict it's probably ok if the filter itself is non-null, it's just that I thought we should keep the toOrFrom field nullable for now (to allow for alternate filters in the future without needing to change the api?).